kafka-rest/src/test/java/io/confluent/kafkarest/integration/ConsumerJsonTest.java

```diff
@@ -114,4 +122,87 @@ public void testConsumeOnlyValues() {
         JsonConsumerRecord::toConsumerRecord);
     commitOffsets(instanceUri);
   }
+
+  @Test
+  public void testConsumeWithMultipleParallelConsumers() throws InterruptedException {
+    class ConsumerTask implements Callable<Void> {
+
+      private int index = 0;
+      private CountDownLatch doneLatch;
+      private CountDownLatch firstConsumeDoneLatch;
+      private CountDownLatch otherConsumerReadyLatch;
+
+      public ConsumerTask(
+          int index,
+          CountDownLatch doneLatch,
+          CountDownLatch firstConsumeDoneLatch,
+          CountDownLatch otherConsumerReadyLatch) {
+        this.index = index;
+        this.doneLatch = doneLatch;
+        this.firstConsumeDoneLatch = firstConsumeDoneLatch;
+        this.otherConsumerReadyLatch = otherConsumerReadyLatch;
+      }
+
+      @Override
+      public Void call() throws InterruptedException {
+        log.info("Consumer instance {}, begin.", index);
+        String instanceUri =
+            startConsumeMessages(
+                groupName, topicName, EmbeddedFormat.JSON, Versions.KAFKA_V2_JSON_JSON, "earliest");
+        consumeMessages(
+            instanceUri,
+            new ArrayList<>(),
+            Versions.KAFKA_V2_JSON_JSON,
+            Versions.KAFKA_V2_JSON_JSON,
+            jsonConsumerRecordType,
+            /* converter= */ null,
+            JsonConsumerRecord::toConsumerRecord);
+        firstConsumeDoneLatch.countDown();
```

Comment:
what does consumeMessages do? I'm wondering if we actually need the latch.  Because we really just need two consumer instances to call poll to enter the rebalance state.

Reply:
@philipnee I renamed the latches, for better readability.

I am unable to repro the exact bug reliably in the integration test. So i did the next best thing, added a general purpose parallel-consume test that repros the bug 'sometimes'. Idea is to setup 2 consumer that subscribe initially, and then do  consume in parallel in a loop.

So the ConsumerTask thread does this -
1. startConsumeMessages - subscribe, consume once, return.
2. signal to other consume that it's `ready to consume in a loop`.
3. wait for other consumer to signal it's `ready for parallel consume`.
4. consumeMessages(20 times in a loop) - this method simply does consume(poll) & return.